Test.c

```diff
@@ -4,11 +4,156 @@
 // Ayisha Sameera,GET.
 // ------------------------------------------------------------------
 // Test.c
-// Program on branch main
+// Program on branch A4.
+// Program implements test cases, gets user input and checks if its a palindrome
+// Prints if the given input string is a palindrome or if reversed INT overflows
+// Prints the reversed integer if the input is within INT range
 // ------------------------------------------------------------------------------------------------
 
+#pragma warning(disable:4996)
 #include <stdio.h>
+#include <stdbool.h>
+#include "Header.h"
+#include <string.h>
+#include <stdlib.h>
+#include <limits.h>
+#include <conio.h>
+#include <malloc.h>
+#include <ctype.h>
 
-int main () { 
-	printf ("Hello World!\n");
+typedef struct {
+   char* str;
+   bool isPal;
+}Type;
+
+char* Filter (char* input, bool* isNumCpy) {
+   // Variable num is set to 1 assuming the input is an integer.
+   int len = (int)strlen (input), index = 0, space = 0;
+   bool isNum = true;
+   char* str = malloc (sizeof (char) * (len + 1));
+   if (str == NULL) return ERROR_MEM_ALLOC;
+   // Negative symbol in first index position of input is included in the filtered string.
+   if (input[0] == '-') str[index++] = '-';
+   for (int k = index; k < len; k++) {
+      if (isdigit (input[k]) == 0) isNum = false;// Check if the input character is a digit.
+      char ch[2] = { input[k],'\0' };
+      // If the character isn't a punctuation , include character in string.
+      if (strpbrk (" \'?,.-\"!;:/`~(){}[]", ch) == NULL)
+         str[index++] = tolower (input[k]);
+      else if (isspace (input[k])) space++;
+   }
+   str[index] = '\0';
+   // Returns EMPTY if the input consists exclusively of space characters.
+   if (space == len) return EMPTY;
+   // If the input string consists solely of punctuation characters, returns the input.
+   if (index == 0) return input;
+   if (isNum == true) {
+      long long int strToNum = atoll (str);
+      // For integer overflow conditions the input is taken as a string i.e isNum = false.
+      if (index > 12 || strToNum > INT_MAX || strToNum < INT_MIN) isNum = false;
+   }
+   if (isNumCpy != NULL) *isNumCpy = isNum;
+   return str;
+}
+
+char* NumToString (int num, char* input) {
+   char* strNum = malloc (sizeof (char) * 12);
+   if (strNum == NULL) strNum = ERROR_MEM_ALLOC;
+   else if (num == -1 && strcmp (input, "-1") != 0) strNum = "Overflow";
+   else sprintf (strNum, "%d", abs (num));
+   return strNum;
+}
+
+void PrintResult (char* str) {
+   bool* isNum = false;
+   char* input = Filter (str, &isNum);                                     // Filtered input string
+   if (!strcmp (str, "\\n") || !strcmp (str, "\\t") || !strcmp (str, "\\r") || !strcmp (input, EMPTY))
+      printf ("Output =  %s\n", EMPTY"\n"PAL_FALSE);
+   else if (isNum == true) {

```

Comment:
if () takes boolean, isNum is also boolean. Please make this change wherever applicable.
if (isNum) is sufficient in most cases.